apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/PartDEventTransformerTest.java

```diff
@@ -40,7 +40,23 @@ public void transformSampleARecord() throws FHIRException {
             .get();
 
     ExplanationOfBenefit eob = PartDEventTransformer.transform(new MetricRegistry(), claim);
-    assertMatches(claim, eob);
+    assertPartDMatches(claim, eob, "01", "01", TransformerConstants.CODING_NPI_US);
+  }
+
+  @Test
+  public void transformSampleARecordWithUPIN() throws FHIRException {
+    List<Object> parsedRecords =
+        ServerTestUtils.parseData(
+            Arrays.asList(StaticRifResourceGroup.SAMPLE_A_UPIN.getResources()));
+    PartDEvent claim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof PartDEvent)
+            .map(r -> (PartDEvent) r)
+            .findFirst()
+            .get();
```

Comment:
After extracting this object, I think you can modify it's values to be whatever you want, which would allow you to remove all of the extra sample data you've had to add here. (See other comments related to this.)